### DISCUSSION TITLE
Private API calls

### DISCUSSION THREAD

**Discussion by mnxt2022** [nuxt/nuxt#10187] [Questions (Nuxt 2)]
Hi everyone, 

Is it possible to make all API calls on server side even after rendering ?

The problem is that we don't want the frontend client sending requests to the external API or make the nuxt server doing it if it is not for navigation/interaction purpose.

Thanks

**Comment by danielroe** [comment]
You could check out [`serverMiddleware`](https://nuxtjs.org/docs/configuration-glossary/configuration-servermiddleware/) which may solve your use-case.

**Comment by CanerSezgin** [comment]
I have similar issue/question which I have described in detail on [stackoverflow](https://stackoverflow.com/questions/70592030/nuxt-server-side-fetch-internal-links-best-practices). I hope you don't mind to ask here and share the external link.

Simply I would like to hide requests/response which they can be seen on network tab as long as it is sent from client. 
When I use asyncData or fetch when the page first time loaded, request sends from server (this is what I'm looking for) however when I click an internal  `nuxt-link` link, it sends request client side. In order to prevent this I tried to replace `<a> tag` instead `nuxt-link` this time it works but after clicking 2 consecutive links, browser freeze and not responding.

Also tried `serverMiddleware` but still it is called from server when the page is loaded first time, internal links never hit the serverMiddleware?
